calculator/main.py

```diff
@@ -26,7 +29,29 @@ def perform_operation(calc, operation, params):
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
     """
-    pass
+    
+    # Test parameters, raise error if invalid
+    if operation != 'plot':
+        valid_params = all(
+                [isinstance(param, numbers.Number) for param in params]
```

Comment:
On the one hand, I think this is a really clever solution. Checking that the parameters are numbers with numbers.Number is smart. Just to suggest an alternative - you could use a try and except and catch all TypeErrors. That way you don't have to import the numbers module and, if someone for whatever reason decided to create a function that takes strings as arguments, you would be able to avoid giving a false negative. But again, given the scope of the problem/project, I really like your approach too.